felix/iptables/match_builder.go

```diff
@@ -343,3 +347,29 @@ func PortRangessToMultiport(ports []*proto.PortRange) string {
 	portsString := strings.Join(portFragments, ",")
 	return portsString
 }
+
+// vniMatchBPF returns a cBPF socket filter program which matches on the VXLAN
+// Network ID of encapsulated packets. The program assumes that it will only be
+// applied to VXLAN UDP datagrams. It is compatible with both IPv4 and IPv6.
+func matchVXLANVNI(vni uint32) []bpf.RawInstruction {
+	asm, err := bpf.Assemble([]bpf.Instruction{
+		// Load offset of UDP payload into X.
+		bpf.LoadExtension{Num: bpf.ExtPayloadOffset}, // ld poff
+		bpf.TAX{}, // tax
```

Comment:
```suggestion
		bpf.TAX{}, // Copy X into A
```
Wasn't familiar with this op so worth spelling it out.